ext/debian/pxp-agent.init

```diff
@@ -0,0 +1,113 @@
+#! /bin/sh
+### BEGIN INIT INFO
+# Provides:          pxp-agent
+# Required-Start:    $network $named $remote_fs $syslog
+# Required-Stop:     $network $named $remote_fs $syslog
+# Should-Start:      pxp-agent
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+### END INIT INFO
+
+PATH=/opt/puppetlabs/puppet/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+DAEMON=/opt/puppetlabs/puppet/bin/pxp-agent
+DAEMON_OPTS="--daemonize "
+PROCNAME="pxp-agent"
+DESC="PXP agent"
+PIDFILE="/var/run/puppetlabs/${PROCNAME}.pid"
+
+test -x $DAEMON || exit 0
+
+[ -r /etc/default/pxp-agent ] && . /etc/default/pxp-agent
+
+. /lib/lsb/init-functions
+
+reload_pxp_agent() {
+    start-stop-daemon --stop --quiet --signal HUP --pidfile $PIDFILE --name $PROCNAME
+}
+
+start_pxp_agent() {
+    start-stop-daemon --start --quiet --pidfile $PIDFILE --startas $DAEMON $DAEMON_OPTS $PXP_AGENT_EXTRA_OPTS
+}
+
+stop_pxp_agent() {
+    start-stop-daemon --stop --retry TERM/10/KILL/5 --quiet --oknodo --pidfile $PIDFILE --name $PROCNAME
+    rm -f "$PIDFILE"
+}
+
+restart_pxp_agent() {
+    log_begin_msg "Restarting $DESC"
+    stop_pxp_agent
+    start_pxp_agent
+    log_end_msg $?
```

Comment:
We don't ever check to see if the stop was successful here. Do we care? Should the restart stop if the stop failed?